### PR TITLE
dev/database: Add test & refactor SQLiteConnection.java

### PR DESCRIPTION
**Pull request by sgoudham**

### CODE REVIEW

**Review by AlanRobinson1337 — APPROVED** [review]
Solid testing

File: src/main/java/org/somerville/swag/data/source/SQLiteConnection.java [hunk 1]
```diff
@@ -10,43 +8,55 @@
...
     private String databaseUrl = JDBC_URL + DATABASE_URL;
 
-    private static final LoggingService loggingService = LoggingServiceImpl.getInstance();
+    private LoggingService loggingService = LoggingServiceImpl.getInstance();
 
-    private SQLiteConnection() throws DatabaseException {
-        connect(databaseUrl);
+    private SQLiteConnection() { }
+
+    public static SQLiteConnection getInstance() {
```
Comment (RobertRanson) [inline src/main/java/org/somerville/swag/data/source/SQLiteConnection.java:23]: Excellent use of the singleton design pattern

File: src/test/java/org/somerville/swag/data/source/SQLiteConnectionTest.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    @Mock
+    LoggingService loggingService;
+
+    @BeforeEach
+    void init_mocks() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    void sqliteConnectionShouldReturnSameInstance() {
```
Comment (RobertRanson) [inline src/test/java/org/somerville/swag/data/source/SQLiteConnectionTest.java:28]: Good test for checking the singleton returns the same instance!

File: src/test/java/org/somerville/swag/data/source/SQLiteConnectionTest.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        String expectedDatabaseUrl = "jdbc:sqlite:" + expectedDatabasePath + expectedDatabaseName;
+        String expectedSuccessMessage = "Successful Connection to Database: " + expectedDatabaseUrl;
+
+        SQLiteConnection sqLiteConnection = SQLiteConnection.getInstance();
+        sqLiteConnection.setLoggingService(loggingService);
+        sqLiteConnection.setDatabaseUrl(expectedDatabaseUrl);
+
+        sqLiteConnection.connect();
+
+        verify(sqLiteConnection.getLoggingService(), times(1)).logDatabaseConnectSuccess(expectedSuccessMessage);
```
Comment (RobertRanson) [inline src/test/java/org/somerville/swag/data/source/SQLiteConnectionTest.java:48]: Checking the logs to ensure a DB connection is a very smart way of testing!

**Review by RobertRanson — APPROVED** [review]
Excellent testing, very robust

**Review by hzl616 — APPROVED** [review]
The database connection testing is coming along nicely. Well spotted with amending the databaseUrl path

**Pull request merged**